File: .idea/SortColors.java
class Solution {
    public void sortColors(int[] nums) {
        if(nums==null || nums.length==0)
            return;
        int left = 0;
        int mid =0;
        int right = nums.length - 1;
        while(mid <= right)
        {
            if(nums[mid]==0)
            {
                swap(nums, mid,left);
                left++;
                mid++;
            }
            else if(nums[mid]==1)
            {
                mid++;
            }
            else if(nums[mid]==2)
            {
                swap(nums, mid, right);
                right--;
            }
        }
    }
    private void swap(int[] nums, int i, int j)
    {
        int temp = nums[i];
        nums[i]=nums[j];
        nums[j]=temp;
    }
}

//time complexity is O(N)
//space complexity is O(1)

File: sort.colors.py
#Time complexity: O(N)
#space COmplexity:O(1)

class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if nums == None or len(nums)==0:
            return
        
        n=len(nums)
        left=0
        right=n-1
        mid=0

        while mid<=right:
            if nums[mid]==2:
                nums[mid],nums[right]=nums[right],nums[mid]
                right -=1
            elif nums[mid]==0:
                nums[mid],nums[left]=nums[left],nums[mid]
                left +=1
                mid +=1
            else:
                mid +=1
        

        
        